packages/babel-helpers/src/helpers.js

```diff
@@ -1011,3 +1011,12 @@ helpers.classPrivateFieldSet = () => template.program.ast`
     return value;
   }
 `;
+
+helpers.classStaticPrivateFieldBase = () => template.program.ast`
+  export default function _classStaticPrivateFieldBase(receiver, classConstructor, privateClass) {
+    if (receiver !== classConstructor && receiver.constructor !== classConstructor) {
```

Comment:
The `receiver.constructor !== classConstructor` shouldn't be necessary.

Reply:
Unfortunately, this seems necessary to be spec compliant. We should discuss that on slack.